src/algorithm/verifySecureUriClaim.ts

```diff
@@ -9,7 +9,10 @@ import { SecureUriClaimVerificationError } from "../error/SecureUriClaimVerifica
  * See also: https://solid.github.io/solid-oidc/#security-tls & https://github.com/solid/authentication-panel/issues/114#issuecomment-751867437
  */
 export function verifySecureUriClaim(uri: string, claim: string): void {
-  if (!uri.startsWith("https://") && !uri.startsWith("http://localhost:")) {
+  if (
+    !uri.startsWith("https://") &&
+    !/^http:\/\/([\w-]+.)*localhost:.*/.test(uri)
+  ) {
```

Comment:
This looks allright to me but I'm arguably not looking at regex very often and I'd like a second pair of eyes on it.

@RubenVerborgh and @joachimvh does this looks good to you as well?

Reply:
Seems correct. Does not cover all possible cases but it might get annoying to write a complete regex that matches all possibilities.

Reply:
No, this regex is unsafe. The `(… .)*` makes it match everything: `/^http:\/\/([\w-]+.)*localhost:.*/.test('http://evil.com/whatever?localhost:')` gives `true`.

Let's not risk anything and parse the URL properly; and check for the `https:` protocol.

Reply:
Oh, I see my mistake, I should have escaped that dot. Would it be considered if I fix the regex?